src/phpDocumentor/Descriptor/FileDescriptor.php

```diff
@@ -42,16 +38,16 @@ class FileDescriptor extends DescriptorAbstract implements Interfaces\FileInterf
     /** @var Collection $constants */
     protected $constants;
 
-    /** @var Collection $functions */
+    /** @var Collection<FunctionDescriptor> $functions */
```

Comment:
:thinking: Why do you type-hint against the concrete classes and not the interfaces? I am not saying that the interfaces are rather useful at this point (there were plans in the past) but didn't phpstan like that?

Reply:
Either I found the concrete class more easily or the interface didn't cover everything. For now PHPStan don't raise error for that. Maybe on higher levels...

Reply:
Let's fix that in a followup; that we either remove the interfaces (if they are useless) or change it so we use those interfaces here

Reply:
created issue https://github.com/phpDocumentor/phpDocumentor/issues/2247 to check this